src/NetAnalyzers/VisualBasic/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/BasicOverrideEqualsOnOverloadingOperatorEquals.vb

```diff
@@ -38,30 +38,38 @@ Namespace Microsoft.CodeQuality.VisualBasic.Analyzers.ApiDesignGuidelines
 
             context.RegisterSymbolAction(
                 Sub(symbolContext)
-                    Dim method = DirectCast(symbolContext.Symbol, IMethodSymbol)
-                    Debug.Assert(method.IsDefinition)
+                    Dim type = DirectCast(symbolContext.Symbol, INamedTypeSymbol)
 
-                    Dim type = method.ContainingType
                     If type.TypeKind = TypeKind.Interface OrElse type.IsImplicitClass OrElse type.SpecialType = SpecialType.System_Object Then
                         ' Don't apply this rule to interfaces, the implicit class (i.e. error case), or System.Object.
                         Return
                     End If
 
-                    ' If there's a = operator...
-                    If method.MethodKind <> MethodKind.UserDefinedOperator OrElse
-                        Not CaseInsensitiveComparison.Equals(method.Name, WellKnownMemberNames.EqualityOperatorName) Then
-
+                    ' ...search for a corresponding Equals override.
+                    If type.OverridesEquals() Then
                         Return
                     End If
 
-                    ' ...search for a corresponding Equals override.
-                    If type.OverridesEquals() Then
+                    ' If there's a = operator...
+                    If Not HasEqualityOperator(type) Then
                         Return
                     End If
 
                     symbolContext.ReportDiagnostic(type.CreateDiagnostic(Rule))
                 End Sub,
-                SymbolKind.Method)
+                SymbolKind.NamedType)
         End Sub
+
+        Private Shared Function HasEqualityOperator(type As INamedTypeSymbol) As Boolean
+            For Each method In type.GetMembers().OfType(Of IMethodSymbol)
```

Comment:
💡 Would be better to iterate over `GetMembers()` and have the `OfType<IMethodSymbol>()` check in the body.